src/recon.rs

```diff
@@ -476,19 +474,16 @@ fn get_lo_ctx(
     let level = |y, x| levels[y * stride + x] as usize;
 
     let mut mag = level(0, 1) + level(1, 0);
-    let offset = match tx_class {
-        TxClass::TwoD => {
-            mag += level(1, 1);
-            *hi_mag = mag as c_uint;
-            mag += level(0, 2) + level(2, 0);
-            ctx_offsets.unwrap()[cmp::min(y, 4)][cmp::min(x, 4)] as usize
-        }
-        TxClass::H | TxClass::V => {
-            mag += level(0, 2);
-            *hi_mag = mag as c_uint;
-            mag += level(0, 3) + level(0, 4);
-            26 + if y > 1 { 10 } else { y * 5 }
-        }
```

Comment:
I already did this optimization in #1241.  As for the `#[inline]`, could you check if it's still bad for perf on top of #1257?

Reply:
It looks like even on top of main, inlining is now ok here. I'll close this PR.

Reply:
Okay, that's good.  @fbossen and I spent a while optimizing it.